### PR TITLE
fix node.labels leak in scheduler predicates e2e

### LINKED ISSUE #26698: PodAffinity SchedulerPredicates are not well isolated
Those tests add labels to Nodes and depend on Node labels, but they don't clean up after themselves, thus leaking some state between tests. This is very bad and fixing this is critical before we can enable tests (and thus pod affinity feature) again.

cc @davidopp @wojtek-t @fejta

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
closes #26698 

cc @davidopp @wojtek-t @fejta @gmarek

### CODE REVIEW

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -529,6 +528,14 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
 		labelPod, err := c.Pods(ns).Get(labelPodName)
 		framework.ExpectNoError(err)
 		Expect(labelPod.Spec.NodeName).To(Equal(nodeName))
+
+		By("removing the label " + k + " off the node " + nodeName)
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: can you wrap this into a function?

File: test/e2e/scheduler_predicates.go [hunk 2]
```diff
@@ -529,6 +528,14 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
...
+		framework.RunKubectlOrDie("label", "nodes", nodeName, k+"-")
+		By("verifying the node doesn't have the label " + k)
+		output := framework.RunKubectlOrDie("describe", "node", nodeName)
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: why kubectl?

**Comment by ixdy** [comment]
I agree with both of @gmarek's comments.

File: test/e2e/scheduler_predicates.go (hunk 1 above)
Comment (ixdy) [inline test/e2e/scheduler_predicates.go]: probably also worth doing this in a `defer`, as otherwise it will only run on success.

Comment (ixdy) [inline test/e2e/scheduler_predicates.go]: (or maybe in `afterEach`?)

**Comment by kevin-wangzefeng** [comment]
@k8s-bot e2e test this please, issue #23545

File: test/e2e/scheduler_predicates.go [hunk 3]
```diff
@@ -125,6 +125,24 @@ func cleanupPods(c *client.Client, ns string) {
 	}
 }
 
+func removeLabelOffNode(c *client.Client, nodeName string, labelKey string) {
+	By("removing the label " + labelKey + " off the node " + nodeName)
+	framework.RunKubectlOrDie("label", "nodes", nodeName, labelKey+"-")
```
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Can you please do ordinary update? Let's not depend on kubectl in this test.

Comment (kevin-wangzefeng) [inline test/e2e/scheduler_predicates.go]: Oops, this line should be removed.
The lines following are in the way ordinary update.

**Comment by yujuhong** [comment]
@k8s-bot node e2e test this issue: #26431

**Comment by kevin-wangzefeng** [comment]
@gmarek  squashed, PTAL

**Comment by davidopp** [comment]
LGTM

Thanks!

I'll let @gmarek apply the LGTM label if he's satisfied with the changes

**Comment by davidopp** [comment]
@k8s-bot unit test this issue: #27205

**Pull request merged**